### PR TITLE
ethernet: Add 2.5G and 5G Ethernet defines

### PR DESCRIPTION
**Pull request by go2sh**
Add required defines for 2.5G and 5G ethernet standards.

Will be used in conjunction with the xgmac driver.

### CODE REVIEW

File: include/zephyr/net/ethernet.h [hunk 1]
```diff
@@ -204,6 +204,12 @@ enum ethernet_hw_caps {
 
 	/** TX-Injection supported */
 	ETHERNET_TXINJECTION_MODE	= BIT(20),
+
+	/** 2.5 Gbits link supported */
+	ETHERNET_LINK_2_5GBASE_T	= BIT(21),
+
+	/** 5 Gbits link supported */
+	ETHERNET_LINK_5GBASE_T		= BIT(22),
```
Comment (pdgendt) [inline include/zephyr/net/ethernet.h:212]:
```suggestion
	ETHERNET_LINK_2500BASE_T	= BIT(21),

	/** 5 Gbits link supported */
	ETHERNET_LINK_5000BASE_T	= BIT(22),
```

Seems to be more in line with other types.

**Comment by carlescufi** [comment]
Thanks for the PR! Please take a look at [our commit message guidelines](https://docs.zephyrproject.org/latest/contribute/guidelines.html#commit-message-guidelines) to find out how to format your commit messages, and at [our contribution workflow](https://docs.zephyrproject.org/latest/contribute/guidelines.html#contribution-workflow) to understand how to update your Pull Request.

File: include/zephyr/net/phy.h [hunk 2]
```diff
@@ -31,17 +31,21 @@ extern "C" {
 /** @brief Ethernet link speeds. */
 enum phy_link_speed {
 	/** 10Base-T Half-Duplex */
-	LINK_HALF_10BASE_T		= BIT(0),
+	LINK_HALF_10BASE_T  	= BIT(0),
```
Comment (jukkar) [inline include/zephyr/net/phy.h:34]: Looks like unnecessary change, why do we need to change the white space here?

File: include/zephyr/net/phy.h [hunk 3]
```diff
@@ -51,7 +55,7 @@ enum phy_link_speed {
  *
  * @return True if link is full duplex, false if not.
  */
-#define PHY_LINK_IS_FULL_DUPLEX(x)	(x & (BIT(1) | BIT(3) | BIT(5)))
+#define PHY_LINK_IS_FULL_DUPLEX(x)	(x & (BIT(1) | BIT(3) | BIT(5) | BIT(6) | BIT(7)))
```
Comment (jukkar) [inline include/zephyr/net/phy.h:58]: unrelated change, please put to separate commit

File: subsys/net/lib/shell/iface.c [hunk 4]
```diff
@@ -56,6 +56,10 @@ static struct ethernet_capabilities eth_hw_caps[] = {
 	EC(ETHERNET_HW_FILTERING,         "MAC address filtering"),
 	EC(ETHERNET_DSA_SLAVE_PORT,       "DSA slave port"),
 	EC(ETHERNET_DSA_MASTER_PORT,      "DSA master port"),
+	EC(ETHERNET_TXTIME,               "TXTIME supported"),
+	EC(ETHERNET_TXINJECTION_MODE,     "TX-Injection supported"),
```
Comment (jukkar) [inline subsys/net/lib/shell/iface.c:60]: Unrelated change, please put to different commit

**Review by jukkar — COMMENTED** [review]
Please add `net: ` prefix to commit subject so `net: ethernet: ...`

File: include/zephyr/net/phy.h (hunk 3 above)
Comment (go2sh) [reply include/zephyr/net/phy.h:58]: Linked to the new phy link speeds above.

File: subsys/net/lib/shell/iface.c (hunk 4 above)
Comment (go2sh) [reply subsys/net/lib/shell/iface.c:60]: Put in a seperate commit.

File: include/zephyr/net/phy.h (hunk 2 above)
Comment (go2sh) [reply include/zephyr/net/phy.h]: Removed.

**Comment by go2sh** [comment]
@jukkar @carlescufi Addressed your feedback..

**Review by jukkar — APPROVED** [review]

**Review by pdgendt — APPROVED** [review]

**Review by rlubos — COMMENTED** [review]
Please fix the compliance issue:
```
3: UC4 Commit message body line exceeds max length (77>75): "Add the 2.5G and 5G link speeds as possible capabilities to ethernet drivers."
```

**Comment by jukkar** [comment]
ping @go2sh

**Comment by go2sh** [comment]
@jukkar Done. Sorry for the delay. Some customer topics put away from working on zephyr.

**Review by rlubos — APPROVED** [review]
